src/appleseed/foundation/array/meshalgorithm.cpp

```diff
@@ -0,0 +1,148 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// appleseed.foundation headers.
+#include "foundation/array/applyvisitor.h"
+#include "foundation/array/array.h"
+#include "foundation/array/arrayview.h"
+#include "foundation/array/exception.h"
+#include "foundation/array/meshalgorithm.h"
+
+// Standard headers.
+#include <algorithm>
+
+using namespace std;
+
+namespace foundation
+{
+namespace
+{
+
+class ComputeBBoxVisitor
+{
+  public:
+    ComputeBBoxVisitor()
+    {
+        m_bbox.invalidate();
+    }
+
+    explicit ComputeBBoxVisitor(const AABB3f& bbox)
+      : m_bbox(bbox)
+    {
+    }
+
+    void operator()(const ArrayView<Vector3f>& view)
+    {
+        for(const Vector3f& p: view)
+            m_bbox.insert(p);
+    }
+
+    template <typename T>
+    void operator()(const ArrayView<T>& view)
+    {
+        throw BadArrayTypeException();
+    }
+
+    AABB3f m_bbox;
```

Comment:
Move data members before methods. Maybe you can turn the class into a struct since it's private anyway and everything in it is public?